widgetastic_manageiq/__init__.py

```diff
@@ -94,6 +95,83 @@ def currently_selected(self):
             return None
 
 
+class SectionedBootstrapSelect(BootstrapSelect):
+    """ Refactor to allow for bootstrap select that have sections. """
+
+    SectionOption = namedtuple("SectionOption", ["text", "value", "section_id"])
+    SECTIONS = ".//li[contains(@class, 'dropdown-header')]"
+    OPTIONS = ".//li[contains(@class, 'data-original-index') and contains(@class, 'data-optgroup')]"
+    SECTION_ELEMENTS = './/li[@data-optgroup="{}"]'
+
+    @property
+    def all_sections(self):
+        return [
+            self.Option(self.browser.text(el), el.get_attribute("data-optgroup"))
+            for el in self.browser.elements(self.SECTIONS, parent=self)
+        ]
+
+    @property
+    def all_options(self):
+        return [
+            self.SectionOption(
+                self.browser.text(
+                    self.browser.element(".//span[contains(@class, 'text')]", parent=el)
+                ),
+                el.get_attribute("data-original-index"),
+                el.get_attribute("data-optgroup"),
```

Comment:
These attributes are getting repeated, let's put them in constants where it makes sense.